src/Microsoft.Android.Build.BaseTasks/AsyncTask.cs

```diff
@@ -75,7 +75,7 @@ protected void Complete (System.Threading.Tasks.Task task)
 		{
 			if (task.Exception != null) {
 				var ex = task.Exception.GetBaseException ();
-				LogError (ex.Message + Environment.NewLine + ex.StackTrace);
+				this.LogUnhandledException (TaskPrefix, ex);
```

Comment:
This may well lock up the UI thread if it is called from a Task. This is one of the reasons we have custom versions of all the logging.

Reply:
The lock up might not show up in our uni tests as they are using a Mock Engine.

Reply:
I think this one is thread safe, as we have this [AsyncTask specific extension](https://github.com/dotnet/android-tools/blob/1ea4e35eaf3fe09b9ca8adf0a29b4d9988146338/src/Microsoft.Android.Build.BaseTasks/UnhandledExceptionLogger.cs#L16)? I think this should ultimately use https://github.com/xamarin/Xamarin.Build.AsyncTask/blob/db4ce14dacfef47435c238b1b681c124e60ea1a0/Xamarin.Build.AsyncTask/AsyncTask.cs#L156?

Reply:
ok cool. I was not aware of that extension method. Should we move the method ?